src/engine/strat_engine/mdv.rs

```diff
@@ -63,35 +108,48 @@ impl MetadataVol {
             }
         }
 
-        if let Err(err) = mount(Some(&dev.devnode()),
-                                &mount_pt,
-                                Some("xfs"),
-                                MsFlags::empty(),
-                                None as Option<&str>) {
-            match err {
-                nix::Error::Sys(nix::Errno::EBUSY) => {
-                    // EBUSY is the error returned in this context when the
-                    // device is already mounted at the specified mountpoint.
+        let mut mdv = MetadataVol {
+            dev,
+            mount_pt,
+            is_mounted: false,
+        };
+
+        {
+            let mount = MountedMDV::mount(&mut mdv)?;
+
+            if let Err(err) = create_dir(mount.mount_pt().join(FILESYSTEM_DIR)) {
+                if err.kind() != ErrorKind::AlreadyExists {
+                    return Err(From::from(err));
                 }
-                _ => return Err(From::from(err)),
             }
-        }
 
-        if let Err(err) = create_dir(&mount_pt.join(FILESYSTEM_DIR)) {
-            if err.kind() != ErrorKind::AlreadyExists {
-                return Err(From::from(err));
+            for dir_e in read_dir(mount.mount_pt().join(FILESYSTEM_DIR))? {
+                let dir_e = dir_e?;
```

Comment:
This should get its own function, like:

(The comments are just an explanation of the reasoning).
```
/// Remove temp files from the designated directory.
/// Fails if the directory can not be read.
/// Persists if an individual directory entry can not be read due to an intermittent IO error.
/// Returns a list of error resulting from remove operations. Ideally, this list is empty.
// This informational result can be upgraded or altered as necessary.
fn remove_temp_files(dir) -> EngineResult<Vec<std::io::Error>> {
  let mut remaining = Vec::new();
  for path in read_dir(dir)?
    // If there's an intermittent IO error on one dir_entry it probably doesn't make sense to fail
    // Presumably it's ok to miss a file that may not be a ".temp" file anyway.
    .filter_map(|e| e.ok())
    // Get the path, pronto, it is what we care about
    .map(|e| e.path())
    // Be interested only in the .temp files
    .filter(|p| p.endswith(".temp")) {
      remove_file(path).unwrap_or_else(|e| remaining.push(e))
  } 
  Ok(remaining)
}
```

Reply:
Maybe, but lets limit this PR to changes related to mounting and open another PR for that.

Reply:
Ok. I've opened an issue and assigned it to myself: https://github.com/stratis-storage/stratisd/issues/579.